modules/connectors/ninja_van/karrio/providers/ninja_van/shipment/create.py

```diff
@@ -0,0 +1,67 @@
+
+import typing
+import karrio.lib as lib
+import karrio.core.units as units
+import karrio.core.models as models
+import karrio.providers.ninja_van.error as error
+import karrio.providers.ninja_van.utils as provider_utils
+import karrio.providers.ninja_van.units as provider_units
+
+
+def parse_shipment_response(
+    _response: lib.Deserializable[dict],
+    settings: provider_utils.Settings,
+) -> typing.Tuple[typing.List[models.RateDetails], typing.List[models.Message]]:
+    response = _response.deserialize()
+
+    messages = error.parse_error_response(response, settings)
+    shipment = (
+        _extract_details(response, settings)
+        if "tracking_number" in response
+        else None
+    )
+
+    return shipment, messages
+
+
+def _extract_details(
+    data: dict,
+    settings: provider_utils.Settings,
+) -> models.ShipmentDetails:
+    shipment = None  # parse carrier shipment type from "data"
+    label = ""  # extract and process the shipment label to a valid base64 text
+    # invoice = ""  # extract and process the shipment invoice to a valid base64 text if applies
+
+    return models.ShipmentDetails(
+        carrier_id=settings.carrier_id,
+        carrier_name=settings.carrier_name,
+        tracking_number="",  # extract tracking number from shipment
+        shipment_identifier="",  # extract shipment identifier from shipment
+        label_type="PDF",  # extract shipment label file format
+        docs=models.Documents(
+            label=label,  # pass label base64 text
+            # invoice=invoice,  # pass invoice base64 text if applies
+        ),
+        meta=dict(
+            # any relevent meta
+        ),
+    )
+
+
+def shipment_request(
+    payload: models.ShipmentRequest,
+    settings: provider_utils.Settings,
+) -> lib.Serializable:
+    shipper = lib.to_address(payload.shipper)
+    recipient = lib.to_address(payload.recipient)
+    packages = lib.to_packages(payload.parcels)
+    service = provider_units.ShippingService.map(payload.service).value_or_key
+    options = lib.to_shipping_options(
+        payload.options,
+        package_options=packages.options,
+    )
+
+    # map data to convert karrio model to ninja_van specific type
+    request = None
```

Comment:
Same as explained above. Needs to be implemented to mirror the expected request data for the Cancel Order API endpoint